build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,182 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package, updatePackageJsonFile, PackageJson } from "@fluidframework/build-tools";
+import { PackageCommand } from "../../BasePackageCommand";
+import { ExtractorConfig } from "@microsoft/api-extractor";
+import { CommandLogger } from "../../logging";
+import path from "node:path";
+import { strict as assert } from "node:assert";
+import * as fs from "fs";
+
+/**
+ * Represents a list of package categorized into two arrays
+ */
+interface PackageTypesList {
+	/**
+	 * An array of strings containing package names that have been updated.
+	 */
+	packagesUpdated: string[];
+	/**
+	 * An array of strings containing package names that have not been updated.
+	 */
+	packagesNotUpdated: string[];
+}
+
+const knownDtsKinds = ["alpha", "beta", "public", "untrimmed"] as const;
```

Comment:
I don't think you should refer to these as dtsTypes. Multiple `.d.ts` files being involved here is really an implementation detail of how this system you are making works with API-extractor.

I think it would make more sense to name thins (like this command, and this list) after "releaseTags" and link https://api-extractor.com/pages/tsdoc/doc_comment_syntax/#release-tags for additional docs.

Naming the command something like "selectReleaseTagToPublish" would be much more clear.

Reply:
"release" is a very overloaded term. I don't want to use it for this purpose. This work is explicitly about types only. Nothing related to what we release when, what version the release will have, etc. This tool only changes the types between "flavors" of the types.

Reply:
It also has nothing to do with "publishing," so I don't think setReleaseTagToPublish is a good choice either. Maybe `setPackageTypesField`?